Modules/_pickle.c

```diff
@@ -2054,6 +2055,43 @@ save_float(PicklerObject *self, PyObject *obj)
 }
 
 static int
+_Pickler_write_large_bytes(
+    PicklerObject *self, const char *header, Py_ssize_t header_size,
+    PyObject *payload, Py_ssize_t payload_size)
+{
+    assert(self->output_buffer != NULL);
+    assert(self->write != NULL);
+    PyObject *result;
+
+    // Commit the previous frame
+    if (_Pickler_CommitFrame(self, 0))
+        return -1;
+
+    // Create a new frame dedicated to the large bytes
+    if (_Pickler_Write(self, header, header_size) < 0)
+        return -1;
+    if (_Pickler_CommitFrame(self, payload_size))
+        return -1;
+
+    // Dump the buffer to the file
+    if (_Pickler_FlushToFile(self) < 0)
+        return -1;
+
+    // Reinitialize the buffer for subsequent calls to _Pickler_Write.
```

Comment:
Some of your comments ends with a period, but others do not. Would be better to be uniform.